### ISSUE TITLE
Route A/B should allow weight 0 for alternate backends

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#10279]
It should be possible to go to zero on alternate backends so that an admin can prepare for a traffic rebalance without actually diverting traffic.  It should continue to be disallowed to set zero on the main backend.

Backends with zero should be omitted from the router.

Target for 1.3 because this is a potential API change and I want to get this correct on the first go.

**Comment by smarterclayton** [comment]
@rajatchopra @knobunc came up in some design sessions - I missed this during the review, but want to make sure it is supported.  weight:0 should be the default value on alternate backends, and should be omitted from consideration by the router (whether that means omitting the backend or haproxy doing the right thing if weight is zero, either way).

**Comment by rajatchopra** [comment]
What is the incentive on disallowing zero on the main backend? Let them all be the same, why not.

**Comment by smarterclayton** [comment]
I don't like allowing:

```
to: 
  name: foo
  weight: 0
alternateBackends:
- name: bar
   weight: 1
```

that feels like an abuse of the system.  On the other hand, if we supported "turn off the traffic to a route by allowing weight 0", I could maybe buy the argument for:

```
to:
  name: foo
  weight: 0
```

on its own.  It's similar to `paused` on deployments - send the 503 error to clients.  Easier than deleting the routes (which is fairly expensive today).

I guess just allow zero on all of them and omit / disable backends with 0 weight.

**Comment by liggitt** [comment]
cc @jwforres @spadgett

**Comment by liggitt** [comment]
I could see validation not allowing the primary and all backends to simultaneously be 0 (there has to be at least one non-zero backend)

**Comment by smarterclayton** [comment]
I think the argument is that you should be able to turn off routes.  That's
useful on its own, so if it's useful and no more/less work than adding an
extra validation rule I'd prefer to get both.

**Comment by liggitt** [comment]
Ok, as long as all participants are aware of the div/0 risk

**Comment by smarterclayton** [comment]
Does the request make sense to you as a user of Openshift?  Set my
route to take no traffic?

**Comment by knobunc** [comment]
There's a request from users to that effect:
  https://trello.com/c/kXrvPFdh/13-3-route-parking-traffic-ingress

I'm a little concerned that this lets people make a submarine disabled route and then enable it later to superseded someone else's.  But that may not be a realistic concern given that if users are expected to be hostile, then they would be administratively separated (by domain name, or the like).

**Comment by smarterclayton** [comment]
I would expect weight 0 to include the route in initial calculation
but omit it from actual endpoints being written.  So the first route
would always "win" and it would not be a submarine (well, at least it
would be on the surface).

**Comment by knobunc** [comment]
LGTM

**Comment by smarterclayton** [comment]
Fixed by #10428
